application/backend/app/api/endpoints/training_configurations.py

```diff
@@ -0,0 +1,98 @@
+from app.services.training_configuration_service import TrainingConfigurationService
+
+logger = logging.getLogger(__name__)
+router = APIRouter(prefix="/api/projects/{project_id}/training_configuration", tags=["Training Configuration"])
+
+
+@router.get("")
+def get_training_configuration(
+    training_configuration_service: Annotated[
+        TrainingConfigurationService, Depends(get_training_configuration_service)
+    ],
+    project_id: UUID,
+    model_architecture_id: str | None = None,
+    model_revision_id: UUID | None = None,
+) -> dict:
+    """
+    Get the training configuration for a project.
+
+    If model_architecture_id is provided, returns configuration for that specific model architecture.
+    If model_revision_id is provided, returns configuration for a specific trained model.
+    If neither is provided, returns only general task-related configuration.
+    Note: model_architecture_id and model_revision_id cannot be used together.
+
+    Args:
+        training_configuration_service (TrainingConfigurationService): The training configuration service.
+        project_id (UUID): The unique identifier of the project.
+        model_architecture_id (Optional[str]): The model architecture ID for specific configuration retrieval.
+        model_revision_id (Optional[UUID]): The model revision ID for specific configuration retrieval.
+
+    Returns:
+        TrainingConfiguration: The training configuration details.
+    """
+    try:
+        training_configuration = training_configuration_service.get_training_configuration(
+            project_id=project_id,
+            model_architecture_id=model_architecture_id,
+            model_revision_id=model_revision_id,
+        )
+        return convert_training_configuration_to_rest(config=training_configuration)
+    except ResourceNotFoundError as e:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=str(e))
+    except ValueError as e:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))
+
+
+@router.patch("", status_code=status.HTTP_204_NO_CONTENT)
+def update_training_configuration(
+    training_configuration_service: Annotated[
+        TrainingConfigurationService, Depends(get_training_configuration_service)
+    ],
+    project_id: UUID,
+    training_config_update: dict,
+    model_architecture_id: str | None = None,
+) -> None:
+    """
+    Update the training configuration for a project.
+
+    - If model_architecture_id is provided, updates configuration for that specific model architecture.
+    - If not provided, updates the general task-related configuration.
+    Note: model_architecture_id cannot be used with model_revision_id for updates.
+
+    Request body should contain elements of the configuration hierarchy to update:
+    ```json
+    {
+      "dataset_augmentation_parameters": {...},
+      "training": {...},
+      "evaluation": {...}
+    }
+    ```
```

Comment:
Should not be here, but in the `Body.description`. Check how it's done in the `update_pipeline` endpoint.